test/applyMiddleware.spec.ts

```diff
@@ -51,16 +51,11 @@ describe('applyMiddleware', () => {
     const spy = jest.fn()
     const store = applyMiddleware(test(spy), thunk)(createStore)(reducers.todos)
 
-    // the typing for redux-thunk is super complex, so we will use an as unknown hack
-    const dispatchedValue = (store.dispatch(
-      addTodoAsync('Use Redux')
-    ) as unknown) as Promise<void>
-    return dispatchedValue.then(() => {
-      expect(spy.mock.calls.length).toEqual(2)
-    })
+    await store.dispatch(addTodoAsync('Use Redux'))
```

Comment:
Now that the `store` is typed properly, this `dispatch` call has no issues. The root of the problem has been fixed.